Systems/dc-3-input.xml

```diff
@@ -132,24 +132,42 @@
       <name>m</name>
       <desc>Mixture richer</desc>
       <binding>
-        <command>property-adjust</command>
-        <property>/controls/engines/mixture-all</property>
-        <step type="double">0.5</step>
-        <min type="double">-1</min>
-        <max type="double">1</max>
-        <wrap type="bool">false</wrap>
+          <command>nasal</command>
+          <script>
+            var mixture = getprop("/controls/engines/mixture-all");
+            if (mixture == 1) {
```

Comment:
This won't work - what if mixture is, say, 0.9999999999999? Then none of these branches are hit, and nothing happens at all.

Instead, you need to check for ranges, e.g.:

```
if (mixture > 0.75) {
    setprop("/controls/engines/mixture-all", 0.5);
}
elsif (mixture > 0.25) {
    setprop("/controls/engines/mixture-all", 0);
}
else {
    setprop("/controls/engines/mixture-all", -1);
}
```

Also, are you sure those steps are correct (-1, 0, 0.5, 1, but not -0.5)?